plugins/builds/artifacts/get.js

```diff
@@ -70,6 +71,23 @@ module.exports = config => ({
                             }).text();
                             const manifestArray = manifest.trim().split('\n');
                             const directoryArray = manifestArray.filter(f => f.startsWith(`./${artifact}/`));
+                            let totalSize = 0;
+
+                            // Check file sizes by fetching metadata
+                            for (const file of directoryArray) {
+                                if (file) {
+                                    const fileMetaResponse = await request.head(`${baseUrl}/${file}?token=${token}&type=download`);
```

Comment:
not pressing: for large no of files fetching metadata can be time consuming, we can run in parallel with a small batch size.

Reply:
also, `req.query.dir && req.query.type === 'download'` , we may need schema validation for query params